### ISSUE TITLE
Stored caches marked as found outside c:geo not shown as found on live map

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#10914]
## Bug description 

Several users of our production version have reported over the last weeks, that found caches leak onto the map as not found.
Finally one user made a more detailed description, which might be helpful to reproduce (or identify it as kind of a server issue).

### Steps to reproduce the problem

- Have caches stored you are going to find
- Log these caches as found **outside of c:geo** (e.g. on PC, whatever)
- Start c:geo, open the live map
- Scroll to the position of these caches

### Actual result after these steps 

- Live map is loaded, caches are shown as stored, caches are not shown as found

### Expected result after these steps 

The updated found state should be retrieved via live map and AFAIR we decided and implemented that the found state gets merged into the saved cache (I might be wrong though)

## c:geo version 

2021.05.20

## Reproducible 

Did not try myself, the user is not sure but had this problem over weeks.

## Additional context 

Might also be some server side delay of marking caches as found (we had such cases in the past) or even some kind of temporarary "not logged in" state.

**Comment by eddiemuc** [comment]
I am confused. As far as I know, locally stored caches are always shown with their offline state across whole c:geo, including the map and including the found status. 

Is there a kind of special loophole implementation for the "found" status which I didn't see so far?

If this should be changed (e.g. caches should be refreshed from server) then this should be done for all cache properties, not just the found status

**Comment by dcjkfgdjhd** [comment]
Not sure if it is related, but when I delete all stored caches at the end of a caching day, the next day they are still stored somewhere in c:geo, and when I use the livemap, they are still displayed as they were before I deleted them (e.g. unfound), not as they should now be displayed based on the fresh livemap data (e.g. found).

I seem to have no way to influence this behaviour, but it always end automatically some time later (maybe about 1 day?) and then they get correctly drawn on the livemap.

**Comment by dcjkfgdjhd** [comment]
Addendum: I agree with eddiemuc that as long as the caches are "officially" stored in c:geo, the stored state should always be what is displayed, not what the livemap reports. But as soon as I say "delete", from now on, the livemap data should rule again, and for me this always happens with some strange delay of about 1 day.

**Comment by Narrat** [comment]
> Not sure if it is related, but when I delete all stored caches at the end of a caching day, the next day they are still stored somewhere in c:geo, and when I use the livemap, they are still displayed as they were before I deleted them (e.g. unfound), not as they should now be displayed based on the fresh livemap data (e.g. found).
> 
> I seem to have no way to influence this behaviour, but it always end automatically some time later (maybe about 1 day?) and then they get correctly drawn on the livemap.

IIRC formerly stored caches are kept 72h in a cache.

**Comment by dcjkfgdjhd** [comment]
> IIRC formerly stored caches are kept 72h in a cache.

OK, thanks. That is actually a good feature, but it should not influence the way the caches are displayed on the livemap.

**Comment by murggel** [comment]
> If this should be changed (e.g. caches should be refreshed from server) then this should be done for all cache properties, not just the found status

For me it would be essential to at least have the "active-state" really "live" for the live map. Because the time I stored the caches locally could be some time ago and when I am then out it could happen, that the cache was deactivated or even archived, but I saw that only after searching a while and then refreshing the cache... so if I am on the live-map, I want to see the live-state of the cache.
But I guess this should go into an own issue, because here it's only about found state?

**Comment by eddiemuc** [comment]
If I assume correctly, the "root cause" of this (and maybe also other behaviour in this direction e.g. the "found-state-not-shown-after-delete" reported in another issue) is the storing of deleted caches for 72h. Maybe we have to rethink this feature: if I delete a cache then I basically don't want to see it anywhere in c:geo after this delete. If some "safe-buffer" should be included in case of accidental cache deleted (which I assume the 72h's are for), then we need some sort of "Trash"/"Papierkorb" where you can see those and are able to restore. And this should be the only place where you see those already deleted caches.

**Comment by dcjkfgdjhd** [comment]
Or alternatively, you could simply ignore the stored data for the livemap display, and on the map always display the data that you received as response from the server. When opening such a cache, and there is an obvious disparity (such as, the livemap reports "found" or "disabled" or "archived" while the stored copy does *not* have this property) you could suggest a refresh to the user ("The cache data might be out of date, would you like to refresh?")

**Comment by jakedot** [comment]
> Maybe we have to rethink this feature: if I delete a cache then I basically don't want to see it anywhere in c:geo after this delete. If some "safe-buffer" should be included in case of accidental cache deleted (which I assume the 72h's are for), then we need some sort of "Trash"/"Papierkorb" where you can see those and are able to restore. And this should be the only place where you see those already deleted caches.

That's a good idea. I think this confusion came from the difference of removing the cache from individual list and removing it from c:geo entirely. SInce it's about lists, a "Trash list" would be a logical place for deleted caches.

**Comment by MagpieFourtyTwo** [comment]
> Or alternatively, you could simply ignore the stored data for the livemap display, and on the map always display the data that you received as response from the server.

I don't know the implementation details, so please correct me if I'm wrong, but I thought, caches which are already stored offline, will not be loaded again from the server ... or am I wrong?

So, if these are currently not loaded anymore, and just in case you want to load *every* cache, no matter if saved (and found) or not, this would IMHO significantly increase mobile data consumption and server traffic - even worse as this would be completely wasted traffic in most of the cases, esp. in "daily route" regions, where thousands of caches are already found, and thus are of no interest at all ... so why load them over and over and over again, just to overcome this little glitch of not showing the right state for 72 h ...?

My idea: Why not leave the grace period (and loading behavior) as is, and just add a "deleted" flag which will in live map trigger a reload of the server version of this cache and merge it with the (still) offline stored version? Merging would be essential, IMHO, cause one of the reasons for the grace period is AFAIK to e. g. not immediately lose Personal Notes or Offline Logs, which (accidentally) have not been uploaded to the server before deleting the entire cache ...

TMALSS: Esp. for offline stored found caches I see no need to reload their status from server. And reloading the live state could IMHO safely be limited to caches which have been manually deleted and currently are in the grace period.

**Comment by eddiemuc** [comment]
> I don't know the implementation details, so please correct me if I'm wrong, but I thought, caches which are already stored offline, will not be loaded again from the server ... or am I wrong?

In live map they are loaded again from the server, but only with a "light" data set (same as for non-stored caches shown on the live map). Reason is that the servers don't allow to specify caches to "exclude" when using their APIs, you simply get everything for a given viewport.

However, thr detailed cache data (description, attributes, logs etc) is not loaded from server in map view for any cache - only if you open the details of a non-stored cache.

So in theory, when an already stored cache is "light"-loaded from the server in live map view, one could merge this data to already stored caches without further bandwidth needs. Question is if and for which properties this should be done.

Many people seem to vote for merging just the cache status in this case (found state, disabled/archived state). I would vote for merging everything that we have (it is service-dependent what this is but usually includes e.g. name, type, d/t, favs, ...)

**Comment by eddiemuc** [comment]
> Reason is that the servers don't allow to specify caches to "exclude" when using their APIs, you simply get everything for a given viewport.

To avoid a misunderstanding: the server APIs e.g. of gc.com and oc **do** allow various filter methods beside the viewport which we use (e.g. the global type filter and the map quick setting excludes are considered as far as the APIs support them, and soon the new filter will be considered as well (again as far as the server APIs support it)), but they do **not** allow to specify a list of geocodes where data should not be loaded.

**Comment by MagpieFourtyTwo** [comment]
[...Merging "light" data set with offline stored version]
> I would vote for merging everything that we have (it is service-dependent what this is but usually includes e.g. name, type, d/t, favs, ...)

Me2. Cause IMHO it makes no sense at all to "ignore" the knowledge c:geo already has about a cache (even for the sake of being probably not completely up-to-date), esp. as this data (like e. g. Attributes and Favs) may be needed by e. g. the new Filter anyway - so why re-fetch it from the server if it's already available offline anyway.

Moreover it will most probably make things way easier if there is just one source of information for a cache.
And this way, the offline stored cache data (like dis/enabled/archived or even found state) can even be updated on the fly ...

**Comment by eddiemuc** [comment]
> ...as this data (like e. g. Attributes and Favs) may be needed by e. g. the new Filter anyway ...

Very true, but unfortunately attributes are not part of the "light" set :-(

**Comment by MagpieFourtyTwo** [comment]
Yep, thought so - and meant it as an example why c:geo should use (thus merge) *all* the information it already has.

**Comment by eddiemuc** [comment]
Teammeeting 11.2.24: recommendation to store updated info obtained via live map data on already stored caches. Flag those caches with some sort of "data could be inconsistent, recommend to refresh cache" flag
